app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -173,16 +189,18 @@ public void onPairingButtonClicked(final View view) {
                 break;
             case R.id.pairing_menu_nfc:
                 intent.setComponent(new ComponentName(this, NFCActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+                this.startActivity(intent);
                 break;
             case R.id.pairing_menu_qr:
                 intent.setComponent(new ComponentName(this, QRExchangeKeyActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
```

Comment:
Unnecessary use of `.putExtra`, see [line 177](https://github.com/ericcornelissen/NervousFish/pull/195/files#diff-149538cc1f1f5e4370b6ae93d6489330R177) of this file